app/assets/javascripts/discourse/components/topic-navigation.js.es6

```diff
@@ -109,6 +111,55 @@ export default Ember.Component.extend({
     }
   },
 
+  _panOpenClose(offset, velocity, direction) {
+
+    const maxOffset = parseInt($(".timeline-container").css("height"));
+    direction === "close" ? offset += velocity : offset -= velocity;
+
+    $(".timeline-container").css("bottom", -offset);
+    if(offset > maxOffset) {
+      this._collapseFullscreen();
+    }
+    else if(offset <= 0) {
+      $(".timeline-container").css("bottom", "");
+    }
+    else {
+      setTimeout(() => this._panOpenClose(offset, velocity, direction), 20);
+    }
+  },
+
+  panStart(e) {
+    const center = e.center;
+    const $centeredElement = $(document.elementFromPoint(center.x, center.y));
+    if (!$centeredElement.parents(".timeline-scrollarea-wrapper").length) {
+      this.set("isPanning", true);
+    }
+    else {
+      this.set("isPanning", false);
+    }
+  },
+
+  panEnd(e) {
+    if(!this.get("isPanning")) {
+      return;
+    }
+    this.set("isPanning", false);
+    //calculate whether we should keep open or close...
```

Comment:
if possible I would remove the comment by a private function:

```
_isPanOpened(e) {
  (e.deltaY > 200 && e.velocityY > -0.15) || e.velocityY > 0.15
}
```

and use this, it removes the need for a comment while staying explicit.